distributed_shampoo/distributed_shampoo.py

```diff
@@ -326,6 +330,7 @@ def __init__(
         precision_config: Optional[PrecisionConfig] = None,
         use_protected_eigh: bool = True,
         track_root_inv_residuals: bool = False,
+        experimental_param_to_lr: Optional[Callable[[torch.Tensor], float]] = None,
```

Comment:
We will not want to expose this directly to the user, but create a flag that merges parameter groups that have different `lr`, `betas`, `beta3`, `epsilon`, `momentum`, `dampening`, or `weight_decay`, but share the same fields everywhere else.